### ISSUE TITLE
Flutter Image.network does not load Firebase Storage images when compiled for WASM

### ISSUE THREAD

**Issue by rayliverified** [flutter/flutter#152909]
Demo: https://flutter-firebase-image-wasm.netlify.app

![image](https://github.com/user-attachments/assets/dd53eec5-8d18-48c6-bd15-12b00451867d)

### Steps to reproduce

1. Load an image URL from Firebase Storage. Example: https://firebasestorage.googleapis.com/v0/b/codeless-dev.appspot.com/o/Qv1csB2UnHh6W3B2Xwmp68RXD642%2Fuploads%2Fimages%2Fcf92e490_3800_11ee_a495_c7bf96d67d4d_image_hotel_app_collage_tilt_right_1200x.png?alt=media
2. Load image via Image.network().
3. Run `flutter build web --wasm`.
4. Upload built files to server.
5. Set server headers to support returning Wasm.
>     Cross-Origin-Embedder-Policy = "credentialless"
>     Cross-Origin-Opener-Policy = "same-origin"
6. Open website and see that the Firebase image doesn't load. Instead, an exception is thrown.

### Expected Result
Images from Firebase Storage should load.
For Wasm Web builds, images loaded from other sources such as Cloudflare or Vercel load correctly. Only Firebase Storage images fail to load.

Can reproduce in Chrome, Edge, and Opera.

### Minimal Reproducible Example
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    //

    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: SingleChildScrollView(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              Image.network(
                'https://firebasestorage.googleapis.com/v0/b/codeless-dev.appspot.com/o/Qv1csB2UnHh6W3B2Xwmp68RXD642%2Fuploads%2Fimages%2Fcf92e490_3800_11ee_a495_c7bf96d67d4d_image_hotel_app_collage_tilt_right_1200x.png?alt=media',
                height: 300,
                fit: BoxFit.fitHeight,
                errorBuilder: (context, error, stackTrace) {
                  return SelectableText('$error, StackTrace: $stackTrace');
                },
              ),
              Image.network(
                'https://assets.codelessly.com/images/demo/hotel_app_collage_tilt_right_1200x.png',
                height: 300,
                fit: BoxFit.fitHeight,
              ),
              const Text(
                'You have pushed the button this many times:',
              ),
              Text(
                '$_counter',
                style: Theme.of(context).textTheme.headlineMedium,
              ),
            ],
          ),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}
```

### Actual results

Web Wasm builds fail to load Firebase Storage images.

### Logs

<details open>
<summary>Logs</summary>

```console
ImageCodecException: Image file format (.png) is not supported by this browser's ImageDecoder API.
Image source: https://firebasestorage.googleapis.com/v0/b/codeless-dev.appspot.com/o/Qv1csB2UnHh6W3B2Xwmp68RXD642%2Fuploads%2Fimages%2Fcf92e490_3800_11ee_a495_c7bf96d67d4d_image_hotel_app_collage_tilt_right_1200x.png?alt=media, StackTrace:     at https://66b1a11bbef9a96df6ae49cc--teal-fudge-a8bfab.netlify.app/main.dart.wasm:wasm-function[9010]:0x170b83
```

</details>

### Flutter Doctor output

<details open>
<summary>Doctor output</summary>

```console
[√] Flutter (Channel stable, 3.22.3, on Microsoft Windows [Version 10.0.22631.3880], locale en-US)
    • Flutter version 3.22.3 on channel stable at D:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b0850beeb2 (3 weeks ago), 2024-07-16 21:43:41 -0700
    • Engine revision 235db911ba
    • Dart version 3.4.4
    • DevTools version 2.34.3

[√] Windows Version (Installed version of Windows is version 10 or higher)

[!] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at C:\Users\AppData\Local\Android\sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11609105)
    ! Some Android licenses not accepted. To resolve this, run: flutter doctor --android-licenses

[X] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.7.5)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.7.34202.233
    • Windows 10 SDK version 10.0.22621.0

[√] Android Studio (version 2024.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11609105)

[√] VS Code (version 1.92.0)
    • VS Code at C:\Users\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
       https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (2 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22631.3880]
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 127.0.2651.86

[√] Network resources
    • All expected network resources are available.
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @rayliverified, this has been fixed on the latest `beta` & `master` channels. I tested with your code sample locally on stable, beta & master and only stable reproduced the issue. 

Per our triage policy, I'll be closing this since it has been fixed on the beta/master channels. If you do experience this issue on either of these channels, kindly leave a comment below and I will reopen this. 

Thank you 

| master & beta | stable | 
| -- | -- |
| <img width="945" alt="Screenshot 2024-08-06 at 08 25 58" src="https://github.com/user-attachments/assets/e3907741-f5c9-4b24-895f-b564a57b71f2"> | <img width="945" alt="Screenshot 2024-08-06 at 08 28 28" src="https://github.com/user-attachments/assets/ee397e89-996b-43c2-8a37-8d7d584b3529"> |


<details>
<summary>flutter doctor -v</summary>

```
[!] Flutter (Channel stable, 3.22.3, on macOS 14.5 23F79 darwin-arm64, locale en-GB)
    • Flutter version 3.22.3 on channel stable at /Users/nexus/dev/sdks/flutter
    ! Warning: `flutter` on your path resolves to /Users/nexus/dev/sdks/flutters/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/nexus/dev/sdks/flutter. Consider adding /Users/nexus/dev/sdks/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/nexus/dev/sdks/flutters/bin/dart, which is not inside your current Flutter SDK checkout at /Users/nexus/dev/sdks/flutter. Consider adding /Users/nexus/dev/sdks/flutter/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b0850beeb2 (3 weeks ago), 2024-07-16 21:43:41 -0700
    • Engine revision 235db911ba
    • Dart version 3.4.4
    • DevTools version 2.34.3
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.
```


```
[!] Flutter (Channel beta, 3.24.0-0.2.pre, on macOS 14.5 23F79 darwin-arm64, locale en-GB)
    • Flutter version 3.24.0-0.2.pre on channel beta at /Users/nexus/dev/sdks/flutter_beta
    ! Warning: `flutter` on your path resolves to /Users/nexus/dev/sdks/flutters/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/nexus/dev/sdks/flutter_beta. Consider adding /Users/nexus/dev/sdks/flutter_beta/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/nexus/dev/sdks/flutters/bin/dart, which is not inside your current Flutter SDK checkout at /Users/nexus/dev/sdks/flutter_beta. Consider adding /Users/nexus/dev/sdks/flutter_beta/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7c6b7e9ca4 (7 days ago), 2024-07-30 14:26:44 +0700
    • Engine revision 6e4deceb38
    • Dart version 3.5.0 (build 3.5.0-323.2.beta)
    • DevTools version 2.37.2
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.
```

```
[✓] Flutter (Channel master, 3.24.0-1.0.pre.462, on macOS 14.5 23F79 darwin-arm64, locale en-GB)
    • Flutter version 3.24.0-1.0.pre.462 on channel master at /Users/nexus/dev/sdks/flutters
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fc465348b9 (5 hours ago), 2024-08-05 21:10:56 -0700
    • Engine revision 376a853846
    • Dart version 3.6.0 (build 3.6.0-116.0.dev)
    • DevTools version 2.38.0
```

</details>

**Comment by rayliverified** [comment]
Confirmed fixed with Flutter v3.24.0.

Thank you for validating and great timing on the stable release :)
